test/jdk/java/nio/channels/DatagramChannel/SelectWhenRefused.java

```diff
@@ -49,9 +49,31 @@ public void test() throws IOException {
 
         // datagram sent to this address should be refused
         SocketAddress refuser = new InetSocketAddress(InetAddress.getLocalHost(), port);
+        System.out.println("Refuser is: " + refuser);
 
-        DatagramChannel dc = DatagramChannel.open().bind(new InetSocketAddress(0));
+        DatagramChannel dc = null;
+        for (int i=0; i < MAX_TRIES; i++) {
+            dc = DatagramChannel.open();
+            try {
+                dc.bind(new InetSocketAddress(0));
+            } catch (Throwable t) {
+                dc.close();
+                throw t;
+            }
+            if (((InetSocketAddress)dc.getLocalAddress()).getPort() == port) {
+                if (i < MAX_TRIES - 1) {
+                    System.out.format("Refuser port has been reused by dc: %s, retrying...%n",
```

Comment:
I think this test is a JUnit test, in which case trace messages should probably go to System.err so that they are inlined with other test output rather than in a separate section of the .jtr file.

Reply:
Good point.